lib/logstash/outputs/file.rb

```diff
@@ -295,6 +295,79 @@ def open(path)
     end
     @files[path] = IOWriter.new(fd)
   end
+
+  ##
+  # Bare-bones utility for running a block of code at an interval.
+  #
+  class Interval
```

Comment:
Should we use [stud/interval](https://github.com/search?q=org%3Alogstash-plugins+%22stud%2Finterval%22&type=Code) here ? It seems to be the preferred way to do this type of task.

Reply:
I initially implemented that way, but I ultimately backed out because `Stud::Interval` for a few of reasons:
 - (a) `Stud::Interval` blocks the calling thread, effectively implementing only the inner loop of this `Interval#run`; while I could combine it with `Stud::Task`, doing so was less straight-forward than this simple interval-runner.
 - (b) when stopped, `Stud::Interval` in a `Stud::Task` will interrupt an in-progress flush, potentially corrupting output, while this implementation will allow an in-progress flush to finish normally.
 - (c) `Stud` is a transitive _development_ dependency via `logstash-devutils` and given (a) and (b), I didn't feel it was worth adding the complexity to our dependency graph.

Reply:
I agree we're missing a Stud::Interval that runs on a separate, managed thread, so I'm ok with having this implementation here (potentially we could contribute this back to stud?)

As for corrupting data, I don't think that is possible, was a thread.wakeup will still block on IO (https://ruby-doc.org/core-2.2.0/Thread.html#method-i-wakeup).

Reply:
> As for corrupting data, I don't think that is possible, as a thread.wakeup will still block on IO

You are correct; I went back and re-read the implementation notes on `Stud#stop!` and it appears to wake up the thread in a more sensible manner than I anticipated.